horovod/common/operations.cc

```diff
@@ -1536,6 +1563,19 @@ int horovod_mpi_threads_supported() {
   }
   return horovod_global.mpi_threads_supported ? 1 : 0;
 }
+
+int horovod_finalize() {
+  if (!horovod_global.initialization_done) {
+    return -1;
+  }
+
+  // Sets the sutdown flag and wait for background thread to terminate.
+  if (horovod_global.background_thread.joinable()) {
```

Comment:
If you do this in multiple threads, you could have a funny business.  I suggest making `shut_down` an `std::atomic_flag`, and doing similar checks to `horovod_init()` to make sure this will only happen once.